go.mod

```diff
@@ -0,0 +1,62 @@
+module github.com/docker/swarmkit
+
+go 1.12
+
+require (
+	code.cloudfoundry.org/clock v0.0.0-20180518195852-02e53af36e6c
```

Comment:
Looks like this commit was now tagged as `1.0.0`; https://github.com/cloudfoundry/clock/commit/02e53af36e6c978af692887ed449b74026d76fec

Reply:
Sadly, go modules does not support this. I can leave a comment making it clear what commit it's using, but that's about it. If you want to verify, you can run `go get code.cloudfoundry.org/clock@1.0.0` which will find the tag, but will write the commit to the mod file. Alternatively, edit the mod file to include the correct version (`go mod edit -require=code.cloudfoundry.org/clock@1.0.0`) and then run `go mod tidy` and it will rewrite it.

Reply:
I should also say that I'm hesitant to leave a comment, the next time we update those just get forgotten and end up being out of date.

Reply:
Filed cloudfoundry/clock#8

Reply:
It doesn't support it because it doesn't have a `v` as prefix? That should be a bug in go mod; SemVer does _not_ require the `v` prefix; in fact, the prefix is _invalid_ SemVer; there was a proposal for a `SemVerTag` standard (which required `v` for tags), but that was removed in SemVer 1.0; https://github.com/semver/semver/issues/1#issuecomment-2605236

> ### Is “v1.2.3” a semantic version?
>
> No, “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control. Example: git tag v1.2.3 -m "Release version 1.2.3", in which case “v1.2.3” is a tag name and the semantic version is “1.2.3”.

I see they had some discussion about this in 2015; https://github.com/golang/go/issues/12302#issuecomment-134993195

Reply:
Yes, sadly the `v` is deliberately required (amusingly, one of the reasons `v` was used is because Docker used it). I can't find the discussion right now, but I think only one was used because they didn't want to deal with the case where someone mixes versions and pins the same version to two different commits (eg. `1.0.0` and `v1.0.0` on the next commit). Which do you pick? In the end, I believe they decided it was easier to just support a single standard and let the Go community normalize on that in the same way that gofmt isn't everybody's favorite format, but the consistency is worth it.

Reply:
ugh; ironic, because they "enforce" SemVer, but don't actually support valid SemVer 🤷‍♂